components/gitpod-db/src/typeorm/migration/1613031552073-CodeSyncDB.ts

```diff
@@ -0,0 +1,16 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+import { tableExists } from "./helper/helper";
+
+export class CodeSyncDB1613031552073 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        await queryRunner.query("CREATE TABLE IF NOT EXISTS `d_b_code_sync_resource` (`userId` char(36) NOT NULL, `kind` varchar(255) NOT NULL, `rev` varchar(255) NOT NULL, `created` timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6), PRIMARY KEY(`userId`, `kind`, `rev`)) ENGINE=InnoDB");
```

Comment:
nit: `VARCHAR(255)` feels excessive for kind - maybe something like `CHAR(64)` is sufficient?

Reply:
similar things go for `rev`? Do we have a more precise expectation towards this field?

Reply:
`rev` is uuid, so `char(36)` as a user id?